group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -190,6 +191,11 @@ public GroupCoordinatorService build() {
      */
     private final CoordinatorRuntime<GroupCoordinatorShard, Record> runtime;
 
+    /**
+     * The scheduler that periodically deletes expired group metadata.
+     */
+    private final KafkaScheduler scheduler = new KafkaScheduler(1, true, "group-metadata-expiration-");
```

Comment:
We already have a timer so we should rather use it. Thinking about this, I think that the best would best to actually schedule the expiration for each shard independently in CoordinatorShard::onLoaded. From there, we can access the `CoordinatorTimer`. Would this work?